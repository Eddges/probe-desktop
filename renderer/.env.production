NEXT_PUBLIC_SENTRY_DSN=https://e1eef2aaa6054d94bffc4a648fb78f09@o155150.ingest.sentry.io/1210892
