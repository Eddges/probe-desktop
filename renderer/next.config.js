/* global require, module */
const withSourceMaps = require('@zeit/next-source-maps')
const withBundleAnalyzer = require('@next/bundle-analyzer')({
  enabled: process.env.ANALYZE === 'true'
})

module.exports = withBundleAnalyzer(withSourceMaps({
  webpack: (config, options) => {
    config.target = 'electron-renderer'

    config.node = {
      __dirname: false,
      __filename: false
    }

    config.module.rules.push({
      test: /\.(eot|ttf|woff|woff2|otf)$/,
      use: [
        {
          loader: 'url-loader'
        }
      ]
    })
    if (!options.isServer) {
      config.resolve.alias['@sentry/node'] = '@sentry/browser'
    }
    return config
  },
})
