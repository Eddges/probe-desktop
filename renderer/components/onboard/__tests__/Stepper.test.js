/**
 * @jest-environment jsdom
 */

import { render, cleanup } from "@testing-library/react";
import { theme } from "ooni-components";
import { ThemeProvider } from "styled-components";
import { IntlProvider } from "react-intl";
import renderer from "react-test-renderer";
import English from "../../../../lang/en.json";

import Stepper from "../Stepper";

const HydratedComponent = (
  { mainComponent },
  locale = "en",
  messages = English
) => {
  return (
    <IntlProvider locale={locale} messages={messages}>
      <ThemeProvider theme={theme}>{mainComponent}</ThemeProvider>
    </IntlProvider>
  );
};

describe("Tests from Stepper component", () => {
  afterEach(() => {
    cleanup();
  });
  test("With active index 0", async () => {
    const component = renderer
      .create(<HydratedComponent mainComponent={`${<Stepper activeIdx={0} />}`} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  test("With active index 1", async () => {
    const component = renderer
      .create(<HydratedComponent mainComponent={`${<Stepper activeIdx={1} />}`} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  test("With active index 2", async () => {
    const component = renderer
      .create(<HydratedComponent mainComponent={`${<Stepper activeIdx={2} />}`} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
