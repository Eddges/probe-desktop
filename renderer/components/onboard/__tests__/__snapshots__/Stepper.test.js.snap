// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests from Stepper component With active index 0 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: #f8f9fa;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: #0588cb;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: #0588cb;
  width: 50px;
  height: 3px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
</div>
`;

exports[`Tests from Stepper component With active index 1 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: #f8f9fa;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: #0588cb;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: #f8f9fa;
  width: 50px;
  height: 3px;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: #0588cb;
  width: 50px;
  height: 3px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c1"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
</div>
`;

exports[`Tests from Stepper component With active index 2 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: #f8f9fa;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: #0588cb;
  width: 15px;
  height: 15px;
  border-radius: 15px;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: #f8f9fa;
  width: 50px;
  height: 3px;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: #0588cb;
  width: 50px;
  height: 3px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c1"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
</div>
`;
